client/keys/show.go

```diff
@@ -144,19 +146,18 @@ func runShowCmd(cmd *cobra.Command, args []string) (err error) {
 }
 
 func fetchKey(kb keyring.Keyring, keyref string) (keyring.Info, error) {
+	// firstly check if the keyref is a key name of a key registered in a keyring.
 	info, err := kb.Key(keyref)
+	if err == nil || !sdkerr.AsOf(err, sdkerr.ErrIO, sdkerr.ErrKeyNotFound) {
+		return info, err
+	}
```

Comment:
Can you explain why we need `|| !sdkerr.AsOf(err, sdkerr.ErrIO, sdkerr.ErrKeyNotFound)` ? Maybe a comment would be good here.